bundles/org.openhab.binding.anel/src/main/java/org/openhab/binding/anel/internal/AnelHandler.java

```diff
@@ -0,0 +1,380 @@
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(AnelConfiguration.class);
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // background initialization
+        scheduler.execute(this::initializeConnection);
+    }
+
+    private void initializeConnection() {
+        final AnelConfiguration config2 = config;
+        final String host = config2 == null ? null : config2.hostname;
+        if (config2 == null || host == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Cannot initialize thing without configuration: " + config2);
+            return;
+        }
+        try {
+            final AnelUdpConnector newUdpConnector = new AnelUdpConnector(host, config2.udpReceivePort,
+                    config2.udpSendPort, scheduler);
+            udpConnector = newUdpConnector;
+
+            // establish connection and register listener
+            newUdpConnector.connect(this::handleStatusUpdate, true);
+
+            // request initial state, 3 attempts
+            for (int attempt = 1; attempt <= IAnelConstants.ATTEMPTS_WITH_COMMUNICATION_ERRORS
+                    && state == null; attempt++) {
+                newUdpConnector.send(IAnelConstants.BROADCAST_DISCOVERY_MSG);
+
+                // answer expected within 50-600ms on a regular network; wait up to 2sec just to make sure
+                for (int delay = 0; delay < 10 && state == null; delay++) {
+                    Thread.sleep(200); // wait 10 x 200ms = 2sec
+                }
+            }
+
+            // set thing status (and set unique property)
+            final AnelState state2 = state;
+            if (state2 != null) {
+                updateStatus(ThingStatus.ONLINE);
+
+                final String mac = state2.mac;
+                if (mac != null && !mac.isEmpty()) {
+                    updateProperty(IAnelConstants.UNIQUE_PROPERTY_NAME, mac);
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Device does not respond (check IP, ports, and network connection): " + config);
+            }
+
+            // schedule refresher task to continuously check for device state
+            periodicRefreshTask = scheduler.scheduleWithFixedDelay(this::periodicRefresh, //
+                    0, IAnelConstants.REFRESH_INTERVAL_SEC, TimeUnit.SECONDS);
+
+        } catch (InterruptedException e) {
+
+            // OH shutdown - don't log anything, just clean up
+            dispose();
+
+        } catch (Exception e) {
+
+            logger.debug("Connection to '{}' failed", config, e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR, "Connection to '" + config
+                    + "' failed with " + e.getClass().getSimpleName() + ": " + e.getMessage());
+            dispose();
```

Comment:
If I see correctly this can also be caused by a network failure or socket error. In this case you should try to re-connect.

Reply:
You are right 👍